src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -2224,7 +2483,7 @@ static const int64_t SECS_TO_NS = 1000000000;
 static const int64_t MSECS_TO_MIS = 1000;
 
 /* clock_gettime () is found by configure on Apple builds, but its only present from ios 10, macos 10.12, tvos 10 and watchos 3 */
-#if defined (HAVE_CLOCK_MONOTONIC) && (defined(TARGET_IOS) || defined(TARGET_OSX) || defined(TARGET_WATCHOS) || defined(TARGET_TVOS))
+#if defined (HAVE_CLOCK_MONOTONIC) && (defined(HOST_IOS) || defined(HOST_OSX) || defined(HOST_WATCHOS) || defined(HOST_TVOS))
```

Comment:
Minimum supported macOS version is 10.15 in .NET 6, per https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md#macos.

Reply:
yes HOST_DARWIN would work too, good point. I can change that in a separate cleanup since we have similar code elsewhere.

Regarding removing this altogether, we have https://github.com/dotnet/runtime/issues/58737 which tracks removing it from Android/iOS but last time we tried we hit various issues (https://github.com/dotnet/runtime/pull/62978) so it needs a bit more care.